server/src/main/java/org/opensearch/rest/action/admin/AdminAPIsUtility.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.rest.action.admin;
+
+import org.opensearch.action.admin.cluster.node.tasks.create.CreateTaskRequest;
+import org.opensearch.action.admin.cluster.node.tasks.create.CreateTaskResponse;
+import org.opensearch.action.admin.cluster.node.tasks.unregister.UnregisterTaskRequest;
+import org.opensearch.action.admin.cluster.node.tasks.unregister.UnregisterTaskResponse;
+import org.opensearch.action.support.TimeoutTaskCancellationUtility;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.rest.RestResponse;
+import org.opensearch.tasks.CancellableTask;
+import org.opensearch.tasks.Task;
+
+public class AdminAPIsUtility {
```

Comment:
i think we can remove the utility and move it to the individual Rest actions.
We can probably introduce method in ActionListener#wrap if the response handler is redundant.

Reply:
> i think we can remove the utility and move it to the individual Rest actions.

It will be duplicate code in each of the rest action.

> We can probably introduce method in ActionListener#wrap if the response handler is redundant.

Ack